src/server/game/Maps/Map.h

```diff
@@ -685,8 +685,10 @@ class InstanceMap : public Map
     void Update(const uint32, const uint32, bool thread = true) override;
     void CreateInstanceScript(bool load, std::string data, uint32 completedEncounterMask);
     bool Reset(uint8 method, std::list<uint32>* globalSkipList = nullptr);
-    uint32 GetScriptId() { return i_script_id; }
-    InstanceScript* GetInstanceScript() { return instance_script; }
+    uint32 GetScriptId() const { return i_script_id; }
+    std::string const& GetScriptName() const;
+    InstanceScript* GetInstanceScript() { return instance_data; }
+    InstanceScript const* GetInstanceScript() const { return instance_data; }
```

Comment:
please use `[[nodiscard]]` on all getters that don't cause any side-effect

Reply:
I was reading about it but can't understand what is this `[[nodiscard]]`

Reply:
@claudiodfc C++17 feature, basically marking a method as `[[nodiscard]]` will make the compiler throw an error automatically whenever the result of that method is ignored. So it's useful to mark getter-methods that are supposed to just give you some data without causing any side effect, because whenever you ignore the result of such a method, you are probably doing something wrong ;)

Reply:
Got it, thanks!